src/components/gas/GasSpeedButton.ios.js

```diff
@@ -139,25 +134,50 @@ const GasSpeedButton = ({
     return filteredGasPrices;
   }, [gasPrices, minGasPrice, options]);
 
-  const gasPrice = get(selectedGasPrice, 'txFee.native.value.amount');
+  const gasPrice = get(selectedGasPrice, 'txFee.native.value.display');
   const customGasPriceTimeEstimateHandler = useRef(null);
 
   const [customGasPriceInput, setCustomGasPriceInput] = useState(0);
   const [estimatedTimeValue, setEstimatedTimeValue] = useState(0);
   const [estimatedTimeUnit, setEstimatedTimeUnit] = useState('min');
   const [inputFocused, setInputFocused] = useState(false);
+  const { nativeCurrencySymbol, nativeCurrency } = useAccountSettings();
 
   const defaultCustomGasPrice = Math.round(
     weiToGwei(gasPricesAvailable?.fast?.value?.amount)
   );
-  const defaultCustomGasPriceUsd = get(
+  const defaultCustomGasPriceNative = get(
     txFees?.fast,
-    'txFee.native.value.amount'
+    'txFee.native.value.display'
   );
   const defaultCustomGasConfirmationTime =
     gasPricesAvailable?.fast?.estimatedTime?.display;
 
-  const price = isNaN(gasPrice) ? '0.00' : gasPrice;
+  // Because of the animated number component
+  // we need to trim the native currency symbol
+  // (and leave the number only!)
+  // which gets added later in the formatGasPrice function
+  const price = (isNil(gasPrice) ? '0.00' : gasPrice)
+    .replace(nativeCurrencySymbol, '')
+    .trim();
+
+  const formatGasPrice = useCallback(
+    animatedValue => {
+      // Polygon is way too cheap, so we need to add more decimals
+      if (currentNetwork === networkTypes.polygon) {
+        return `${nativeCurrencySymbol}${(
+          Math.ceil(Number(animatedValue) * 1000000) / 1000000
+        ).toFixed(6)}`;
+      } else {
+        return `${nativeCurrencySymbol}${
+          nativeCurrency === 'ETH'
+            ? (Math.ceil(Number(animatedValue) * 10000) / 10000).toFixed(4)
+            : (Math.ceil(Number(animatedValue) * 100) / 100).toFixed(2)
+        }`;
+      }
+    },
+    [currentNetwork, nativeCurrencySymbol, nativeCurrency]
+  );
```

Comment:
seems like the same logic is on `.android` file, not sure if is something we should care right now but it could be difficult to maintain in the future

Reply:
100% agree, we should consolidate but I don't wanna add more to this PR.